### PR TITLE
Changed some code in presenters & added taking book by picture to con…

### PR DESCRIPTION
**Pull request by neringga**
…troller

### CODE REVIEW

File: VILIB/Controllers/BookController.cs [hunk 1]
```diff
@@ -33,10 +33,28 @@ public HttpResponseMessage Get()
...
+            if (file.ContentLength > 0)
+            {
+                var fileName = Path.GetFileName(file.FileName);
+                var path = Path.Combine(HttpContext.Current.Server.MapPath("~/uploads"), fileName);
+                var decodedCode = _scannerPresenter.DecodedBarcode(path);
+                var book = _bookPresenter.FindBookByCode(decodedCode.Text);
+                var takenBook = _takenBookPresenter.AddTakenBook(book, StaticDataSource.CurrUser);
+                return new HttpResponseMessage
+                {
+                    Content = new StringContent(JsonConvert.SerializeObject(takenBook.HasToBeReturned),
```
Comment (austeja) [inline VILIB/Controllers/BookController.cs]: Is the manual serialization neccessary? cannot you define the serialize method in the Book class itself?

File: VILIB/Controllers/BookController.cs [hunk 2]
```diff
@@ -33,10 +33,28 @@ public HttpResponseMessage Get()
...
+                        System.Text.Encoding.UTF8, "application/json")
+                };
+                //file.SaveAs(path);
+            }
+
+            return new HttpResponseMessage
```
Comment (austeja) [inline VILIB/Controllers/BookController.cs]: Would return an error code that would indicate that something has gone wrong

File: VILIB/Controllers/BookController.cs (hunk 1 above)
Comment (neringga) [reply VILIB/Controllers/BookController.cs]: I didn't think about that. I will fix this

Comment (neringga) [reply VILIB/Controllers/BookController.cs]: I didn't think about that. I will fix this

**Pull request merged**